google/cloud/testing_util/opentelemetry_matchers.h

```diff
@@ -186,6 +194,38 @@ ::testing::Matcher<SpanDataPtr> SpanEventsAre(Args const&... matchers) {
   return SpanEventsAreImpl(::testing::ElementsAre(matchers...));
 }
 
+MATCHER_P(LinkHasSpanContext, context,
```

Comment:
I will wait to see what the tests look like.

If we only ever `LinkHasSpanContext(span->GetContext())`.... then maybe the matcher should be `LinksToSpan(span)`.